python/python/lance/dataset.py

```diff
@@ -689,6 +689,83 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified
         )
 
+    def create_scalar_index(
```

Comment:
I wasn't sure whether we wanted to expose scalar indices as a separate method (`create_scalar_index`) or a different index type.  Internally we have a single method with a different index type.

I ended up opting for two different methods but I could be argued out of it.  My concern is that users will not realize that the one API can do both things and they would need too much sophistication to use it correctly.  I'm hoping exposing this as two different APIs reduces the cognitive load on the user.

Reply:
:shrug: Yeah I'm not sure either way. This approach seems fine.